cincinnati/src/plugins/internal/channel_filter.rs

```diff
@@ -41,6 +42,20 @@ impl ChannelFilterPlugin {
 
         Ok(Box::new(plugin))
     }
+
+    pub fn remove_conditional_edges(
+        &self,
+        ce: &mut Vec<cincinnati::ConditionalEdge>,
+        releases: HashSet<String>,
+    ) -> Fallible<()> {
+        ce.into_iter()
+            .for_each(|ce: &mut cincinnati::ConditionalEdge| {
+                ce.edges
+                    .retain(|e| !releases.contains(&e.from) || !releases.contains(&e.to));
```

Comment:
[De Morgan's laws][1] has `!a || !b = !(a && b)`, and I think the form with fewer `!` is easier to read.  Although in this case, I think you may have the sense flipped.  Shouldn't it actually be `releases.contains(&e.from) && releases.contains(&e.to)` to only retain edges where both the source and target are in the graph?

[1]: https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Reply:
Ah, perhaps the `releases` set being passed in is the set of releases that are being pruned out?  Can we pivot to have it be the set of releases that are being preserved?  With an explicit name like `preserved_releases`?  Because that's a well-defined set based on current graph state, while "releases we should remove" depends on either tracking history or comparing the set of all conditional-edge-mentioned releases with the set of all the current graph's releases.

Reply:
like we are doing to the releases, the graph contains all the releases and all the conditional edges for all available architectures. 
here we are removing the releases and removing the conditional edges for the removed releases. 
this is just an extension of the logic that we are using to remove releases and their edges.